### PR TITLE
Typescript Conversion RFC

### PR DESCRIPTION
**Pull request by williamluke4**
This is a first pass at converting to typescript, tests have not been run yet

### CODE REVIEW

File: dist/endpoints/b2b.d.ts [hunk 1]
```diff
@@ -1,37 +1,38 @@
...
+ * @description - Use mpesa api to transit Mpesa Transaction from one company to another.
  * @param  {number}  senderParty                              Organization Sending the transaction
  * @param  {number}  receiverParty                            Organization Receiving the funds
  * @param  {number}  amount                                   The amount been transacted
  * @param  {string}  queueUrl                                 The path that stores information of time out transactions
  * @param  {string}  resultUrl                                The path that receives results from M-Pesa.
  * @param  {number}  [senderType=4]                           Type of organization sending the transaction[ default Shortcode]
  * @param  {number}  [receiverType=4]                         Type of organization receiving the transaction [ default Shortcode]
- * @param  {string}  [initiator=null]                         This is the credential/username used to authenticate the transaction request.
+ * @param  {string}  [initiator=null]                         mpesa is the credential/username used to authenticate the transaction request.
```
Comment (williamluke4) [inline dist/endpoints/b2b.d.ts]: Revert

**Comment by williamluke4** [comment]
Am I correct in saying that the integration test does not work in the master branch?

**Comment by geofmureithi** [comment]
Tests should pass. EG: https://travis-ci.org/safaricom/mpesa-node-library/builds/461261949
Sometimes Daraja may need tests to be executed again due to unexpected HTTP`500` responses.
Your build does not even contact Daraja.

**Comment by geofmureithi** [comment]
Sadly your changes are also *breaking* which is not desirable at the moment.

```
mpesaApi.c2bRegister(URL + '/c2b/validation', URL + '/c2b/success')
```
is now
```
mpesaApi.c2bRegister(options)
```

I do understand the feeling to use an object, but there are reasons why we decided to go with arguments:

- A lot of Daraja arguments do not necessarily need to be from programmer input, the lib should be able to handle it.
- Look at an API like 
```
const { shortCode } = mpesaApi.configs
const testMSISDN = 254708374149
await mpesaApi.b2c(shortCode, testMSISDN, 100, URL + '/b2c/timeout', URL + '/b2c/success')
```
We dont need shortcode here. we only need 4 args here. I dont foresee us going towards the object direction.

Maybe in version 2 we can use Typescript but then the direction will be different as shown below:
```
export type B2BOptions = {
  senderParty: number;
  receiverParty: number;
  amount: number;
  queueUrl: string;
  resultUrl: string;
  senderType?: number;
  receiverType?: number;
  initiator?: string;
  commandId?: string;
  accountRef?: string;
  remarks?: string;
};
// May be broken down
export type Sender = {
   senderParty: number;
   senderType?: number;
}
```

**Comment by geofmureithi** [comment]
Closing because it's breaking the current API.
Feel free to start an issue for version 2 and then we can first discuss the way forward.

**Comment by geofmureithi** [comment]
Link to v0.2.0 https://github.com/safaricom/mpesa-node-library/milestone/1

**Pull request closed without merging**